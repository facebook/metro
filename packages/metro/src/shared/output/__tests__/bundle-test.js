/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+js_foundation
 * @flow
 */

'use strict';

jest.mock(
  '../../../Server',
  () =>
    class MockServer {
      build() {
        return Promise.resolve({code: 'code', map: '{"ast":"some code"}'});
      }
    },
);

const bundle = require('../bundle');
const Server = require('../../../Server');
const {getDefaultValues} = require('metro-config/src/defaults');
const config = getDefaultValues('/');
const requestOptions = {
  entryFile: 'test',
  minify: false,
  platform: 'ios',
};

it('should succeed', async () => {
  expect(
    await bundle.build(new Server(config), {...requestOptions}),
  ).toMatchSnapshot();
});

it('should call postProcessBundleSourcemap', async () => {
  const postProcessBundleSourcemap = jest.fn(({code, map}) => ({
    code: `modified ${code}`,
    map,
  }));
  expect(
    await bundle.build(new Server(config), {
      ...requestOptions,
      postProcessBundleSourcemap,
    }),
  ).toMatchSnapshot();
  expect(postProcessBundleSourcemap).toBeCalled();
});
