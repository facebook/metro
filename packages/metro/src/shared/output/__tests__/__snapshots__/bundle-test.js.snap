// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call postProcessBundleSourcemap 1`] = `
Object {
  "code": "modified code",
  "map": "{\\"ast\\":\\"some code\\"}",
}
`;

exports[`should succeed 1`] = `
Object {
  "code": "code",
  "map": "{\\"ast\\":\\"some code\\"}",
}
`;
